pymongo/encryption.py

```diff
@@ -682,6 +704,74 @@ def encrypt(
                 query_type=query_type,
                 contention_factor=contention_factor,
+                range_opts=range_options,
+            )
+            return decode(encrypted_doc)["v"]  # type: ignore[index]
+
+    def encrypt_expression(
+        self,
+        expression: Mapping[str, Any],
+        algorithm: str,
+        key_id: Optional[Binary] = None,
+        key_alt_name: Optional[str] = None,
+        query_type: Optional[str] = None,
+        contention_factor: Optional[int] = None,
+        range_options: Optional[EncryptionRangeOpts] = None,
+    ) -> RawBSONDocument:
+        """Encrypt a BSON expression with a given key and algorithm.
+
+        Note that exactly one of ``key_id`` or  ``key_alt_name`` must be
+        provided.
+
+        :Parameters:
+          - `expression`: The BSON aggregate or match expression to encrypt.
+          - `algorithm` (string): The encryption algorithm to use. See
+            :class:`Algorithm` for some valid options.
+          - `key_id`: Identifies a data key by ``_id`` which must be a
+            :class:`~bson.binary.Binary` with subtype 4 (
+            :attr:`~bson.binary.UUID_SUBTYPE`).
+          - `key_alt_name`: Identifies a key vault document by 'keyAltName'.
+          - `query_type` (str): **(BETA)** The query type to execute. See
+            :class:`QueryType` for valid options.
+          - `contention_factor` (int): **(BETA)** The contention factor to use
+            when the algorithm is :attr:`Algorithm.INDEXED`.  An integer value
+            *must* be given when the :attr:`Algorithm.INDEXED` algorithm is
+            used.
+          - `range_options`: An instance of EncryptionRangeOptions.
+
+        .. note:: Support for Range queries is in beta.
+           Backwards-breaking changes may be made before the final release.
+
+        :Returns:
+          The encrypted value, a :class:`~bson.RawBSONDocument`.
+
+        .. versionadded:: 4.4
+        """
+
+        self._check_closed()
+        if key_id is not None and not (
+            isinstance(key_id, Binary) and key_id.subtype == UUID_SUBTYPE
+        ):
+            raise TypeError("key_id must be a bson.binary.Binary with subtype 4")
+
+        doc = encode(
+            {"v": expression},
+            codec_options=self._codec_options,
+        )
+        range_options = encode(
+            range_options.as_doc(),
+            codec_options=self._codec_options,
+        )
+        with _wrap_encryption_errors():
+            encrypted_doc = self._encryption.encrypt(
+                value=doc,
+                algorithm=algorithm,
+                key_id=key_id,
+                key_alt_name=key_alt_name,
+                query_type=query_type,
+                contention_factor=contention_factor,
+                range_opts=range_options,
+                is_expression=True,
             )
             return decode(encrypted_doc)["v"]  # type: ignore[index]
```

Comment:
I think we need to use a raw bson codec_options in order to preserve the BSON here.